Add config properties for S3-compatible storage

zz22394 opened the pull request:
I added two config properties for issue #6053 

Config properties | Details
------------ | -------------
hive.s3.endpoint | Allow endpoint other than AWS official server.
hive.s3.signer-type | Enable specifying Sign type when connecting S3 API.   Possible value is [here](https://github.com/elastic/elasticsearch/pull/13360)

electrum commented:
Can you add these new properties to the Hive connector documentation? (In file hive.rst)

electrum commented:
Let's make SignerType an enum so it's easy to see the allowed values and perform validation when the config is created. For example, see HiveStorageFormat which is used in the Hive config.

zz22394 commented:
@electrum 
1. I added descriptions  for these two new properties to Hive connector documentation.
2. Changed String SignerType to a new enum named ``PrestoS3SignerType``. Because S3SignerType is already been used as an enum type, I added a prefix 'Presto' to avoid confusing.

Would you please have a look at it?

electrum commented on presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java:
```diff
@@ -617,6 +623,11 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
             }
         }
 
+        String endPoint = hadoopConfig.get(S3_ENDPOINT);
```
Nit: "endpoint" is one word so it should be `endpoint`

electrum commented on presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3SignerType.java:
```diff
@@ -0,0 +1,38 @@
+
+import static java.util.Objects.requireNonNull;
+
+public enum PrestoS3SignerType
```
This violates normal enum naming conventions (but is fine in this case), so we should add
```java
@SuppressWarnings("EnumeratedConstantNamingConvention")

electrum commented on presto-docs/src/main/sphinx/connector/hive.rst:
```diff
@@ -174,6 +174,11 @@ Property Name                                      Description
 ``security.config-file``                           Path of config file to use when ``hive.security=file``.
                                                    See :ref:`hive-file-based-authorization` for details.
+
+``hive.s3.endpoint``                               The S3 storage endpoint server.
```
Let's put these with the other S3 configs

electrum replied:
We should mention that this is for S3-compatible storage instead of AWS

electrum commented on presto-docs/src/main/sphinx/connector/hive.rst:
```diff
@@ -174,6 +174,11 @@ Property Name                                      Description
+
+``hive.s3.endpoint``                               The S3 storage endpoint server.
+
+``hive.s3.signer-type``                            Enable specifying Sign type when connecting S3 API.
```
Same here (AWS users shouldn't change it)

electrum commented:
Merged, thanks!
